homeassistant/components/netgear_lte/__init__.py

```diff
@@ -137,90 +146,89 @@ def get_modem_data(self, config):
-    # Load platforms for each modem
-    for lte_conf in netgear_lte_config:
-        # Notify
-        for notify_conf in lte_conf[CONF_NOTIFY]:
-            discovery_info = {
-                CONF_HOST: lte_conf[CONF_HOST],
-                CONF_NAME: notify_conf.get(CONF_NAME),
-                CONF_NOTIFY: notify_conf,
-            }
-            hass.async_create_task(
-                discovery.async_load_platform(
-                    hass, Platform.NOTIFY, DOMAIN, discovery_info, config
-                )
-            )
 
-        # Sensor
-        sensor_conf = lte_conf[CONF_SENSOR]
-        discovery_info = {CONF_HOST: lte_conf[CONF_HOST], CONF_SENSOR: sensor_conf}
-        hass.async_create_task(
-            discovery.async_load_platform(
-                hass, Platform.SENSOR, DOMAIN, discovery_info, config
-            )
-        )
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Netgear LTE from a config entry."""
+    host = entry.data[CONF_HOST]
+    password = entry.data[CONF_PASSWORD]
 
-        # Binary Sensor
-        binary_sensor_conf = lte_conf[CONF_BINARY_SENSOR]
-        discovery_info = {
-            CONF_HOST: lte_conf[CONF_HOST],
-            CONF_BINARY_SENSOR: binary_sensor_conf,
-        }
-        hass.async_create_task(
-            discovery.async_load_platform(
-                hass, Platform.BINARY_SENSOR, DOMAIN, discovery_info, config
-            )
-        )
+    if DOMAIN not in hass.data:
+        websession = async_create_clientsession(hass, cookie_jar=CookieJar(unsafe=True))
 
-    return True
+        hass.data[DOMAIN] = LTEData(websession)
 
+    modem = eternalegypt.Modem(hostname=host, websession=hass.data[DOMAIN].websession)
+    modem_data = ModemData(hass, host, modem)
 
-async def _setup_lte(hass, lte_config):
-    """Set up a Netgear LTE modem."""
+    await _login(hass, entry, modem_data, password)
 
-    host = lte_config[CONF_HOST]
-    password = lte_config[CONF_PASSWORD]
+    await async_setup_services(hass)
 
-    websession = hass.data[DOMAIN].websession
-    modem = eternalegypt.Modem(hostname=host, websession=websession)
+    _legacy_task(hass, entry)
 
-    modem_data = ModemData(hass, host, modem)
+    await hass.config_entries.async_forward_entry_setups(
+        entry, [platform for platform in PLATFORMS if platform != Platform.NOTIFY]
+    )
+
+    return True
 
-    try:
-        await _login(hass, modem_data, password)
-    except eternalegypt.Error:
-        retry_task = hass.loop.create_task(_retry_login(hass, modem_data, password))
 
-        @callback
-        def cleanup_retry(event):
-            """Clean up retry task resources."""
-            if not retry_task.done():
-                retry_task.cancel()
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    loaded_entries = [
```

Comment:
We need to log out from the modem on config entry unload too, not only on Home Assistant stop.